<!-- -->
<launch>
  <arg name="input_obj_topic" default="/detection/lidar_detector/objects" />
  
  <arg name="output_topic" default="/detection/object_tracker/objects" />
  <arg name="zzz_output_topic" default="/zzz/perception/objects_tracked" />

  <arg name="dis_sqrt_thres" default="16.0" />
  <arg name="birth_thres" default="3" />
  <arg name="die_thres" default="3" />
  <arg name="marker_display_duration" default="0.1" />
  <arg name="global_output" default="true" />
  
  <!-- carla odom topic -->
  <arg name="input_odm_topic" default="/carla/ego_vehicle/odometry" />

  <!-- xiaopeng odom topic -->
  <!-- <arg name="input_odm_topic" default="/gps/odom" /> -->

  <arg name="is_benchmark" default="false"/>
  <arg name="kitti_data_dir" default="$(env HOME)/shared_dir/result.txt"/>

  <arg name="namespace" default="/detection/object_tracker/"/>

  <node pkg="global_ukf_track" type="global_ukf_track" name="global_ukf_track_01" output="screen">
    <param name="global_output" value="$(arg global_output)"  />
    <param name="input_odm_topic" value="$(arg input_odm_topic)"  />
    <param name="input_obj_topic" value="$(arg input_obj_topic)"  />
    <param name="output_topic" value="$(arg output_topic)"  />
    <param name="dis_sqrt_thres" value="$(arg dis_sqrt_thres)"  />
    <param name="birth_thres" value="$(arg birth_thres)"  />
    <param name="die_thres" value="$(arg die_thres)"  />
    <param name="is_benchmark" value="$(arg is_benchmark)" />
    <param name="kitti_data_dir" value="$(arg kitti_data_dir)" />
  </node>

  <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="ukf_track_visualization_01"
        output="screen" ns="$(arg namespace)">
        <param name="marker_display_duration" value="$(arg marker_display_duration)"  />
  </node>

  <node pkg="topic_tools" type="relay" name="ukf_track_relay_01"
        output="screen" args="$(arg namespace)/objects /detection/objects"/>

</launch>
